### ISSUE TITLE
Can ipfs-lite be used along the ipfs daemon on the same host?

### ISSUE THREAD

**Issue by akakream** [hsanjuan/ipfs-lite#316]
I am currently using the ipfs-lite along the ipfs daemon on the same host. I have a private swarm for the ipfs daemon that is independent from ipfs-lite's swarm. I communicate between the nodes and share state using [go-ds-crdt](https://github.com/ipfs/go-ds-crdt), which is powered by ipfs-lite as shown in the example in the go-ds-crdt repository. 

However, I observe some strange behaviour in the ipfs daemon when ipfs-lite is running along:
- Some nodes cannot download data from certain other nodes. The node that cannot download data can swarm the other nodes and see them as providers of the data. However, they cannot download the data (neither using the gateway nor using `ipfs get`.)
-  When I stop ipfs-lite, the node can find the data and download it.
- It seems running ipfs-lite along ipfs daemon creates confusion for the nodes in terms of routing, even though ipfs-lite is using in memory datastore and a different dht etc.
- I set up ipfs-lite following the example from [https://github.com/ipfs/go-ds-crdt](https://github.com/ipfs/go-ds-crdt)

I was wondering if this behaviour is expected?

**Comment by hsanjuan** [comment]
That is unexpected... and weird. I suppose you are not sharing the same peerID/key for both things right?

What happens with `ipfs get`, it just hangs without getting any block?

You say that ipfs-lite is in one private swarm and Kubo is running on a different private swarm so they should not conflict at all? Is there a chance that this is your router being unable to handle certain when both things are running? How big are your swarms?

**Comment by akakream** [comment]
- The peerIDs/keys are different for ipfs-lite and kubo
- The swarm size is 4 hosts. All hosts are running both ipfs-lite and Kubo. 
- Kubo has its own private swarm. It uses a swarm key and bootstraps one node. 
- ipfs-lite bootstraps one ipfs-lite peer creating its own swarm.
- So Kubo only knows "Kubo peers" and ipfs-lite only knows "ipfs-lite peers".

I have observed that `ipfs get` sometimes completely hangs and sometimes slows down seriously. When I shut down the ipfs-lite, the download speed gets back to normal and I observe no problem in `ipfs get`.

- All 4 hosts are machines on the Google Cloud platforms. I do not think it is an infrastructure problem.

Since you know the internals of both projects better, can you think of any way that ipfs-lite would interfere with Kubo?

**Comment by akakream** [comment]
I realized that MDNS discovery is enabled in Kubo for all peers. Could it be related to that? 
- Is MDNS discovery enabled for ipfs-lite by default?
- Even though I do not see the ipfs-lite peers when I do `ipfs swarm peers` from Kubo, is it possible that they find each other or pollute each other's DHT's using MDNS?   

I turned it off and doing tests right now. It is still early to confirm that I see a difference.

- Also I use ipfs-lite only to have go-ds-crdt. I do not do any `get` or `add`. Would starting ipfs-lite offline as described [here](https://pkg.go.dev/github.com/dcnetio/ipfs-lite#Config) help? Although, I do not know if starting it as `Offline` hurt go-ds-crdt capabilities.
- Anything else that comes to your mind?
